Il2CppInterop.Generator/Extensions/ILGeneratorEx.cs

```diff
@@ -182,6 +195,12 @@ public static void EmitObjectToPointer(this ILProcessor body, TypeSignature orig
                 body.Add(OpCodes.Conv_I);
             }
         }
+        else if (originalType is PointerTypeSignature)
+        {
+            Debug.Assert(newType is PointerTypeSignature);
+
+            body.AddLoadArgumentAddress(argumentIndex);
```

Comment:
I have concerns about this line. It's slightly different from how blittable value types are handled directly below.